### PR TITLE
feat: drag and drop style update in chats & allow dropping onto contacts in contacts view

### PR DESCRIPTION
**Pull request by lostfictions**
#### Relevant info and issue/PR links

https://app.clubhouse.io/peerio/story/15513/desktop-drag-and-drop-style-update

This PR removes the old `DropTarget` in favour of a unified UI in the style already used for drag and drop in the Files pane.

I've also refactored `ChatView` so that it's three components and `ContactList` so that it's two; all the new components should be testable with Enzyme/whatever (but new tests are not included in this PR).

#### Testing instructions

Most stuff relating to the contact list and uploading files (both drag and drop and via menu) should be tested if possible!

----
### Repository owner

- [ ] Was tested and can be merged.
- [ ] PR name follows conventional changelog format.

[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)

### CODE REVIEW

File: src/ui/contact/ContactList.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+import routerStore from '~/stores/router-store';
+import UploadDialog from '~/ui/shared-components/UploadDialog';
+import T from '~/ui/shared-components/T';
+
+import ContactListItem from './ContactListItem';
+
+@observer
+export default class ContactList extends React.Component {
+    readonly sortOptions = [
+        { value: 'firstName', label: t('title_firstName') },
```
Comment (lostfictions) [inline src/ui/contact/ContactList.tsx]: belatedly realizing this needs to be a getter or it's not going to change when the locale does

**Comment by ebilg** [comment]
commented on CH

**Comment by BlueMona** [comment]
commented on CH

**Comment by BlueMona** [comment]
tested for fix. Ok to merge.

**Pull request merged**